modules/packages.xqm

```diff
@@ -264,22 +264,16 @@ declare function packages:scan-repo($callback as function(xs:string, element(),
 };
 
 (: should be private but there seems to be a bug :)
-declare function packages:get-package-meta($app as xs:string, $name as xs:string) {
-    let $data :=
-        let $meta := repo:get-resource($app, $name)
-        return
-            if (exists($meta)) then util:binary-to-string($meta) else ()
-    return
-        if (exists($data)) then
-            try {
-                parse-xml($data)
-            } catch * {
-                <meta xmlns="http://exist-db.org/xquery/repo">
-                    <description>Invalid repo descriptor for app {$app}</description>
-                </meta>
-            }
-        else
-            ()
+declare function packages:get-package-meta($app as xs:string, $name as xs:string) as element(repo:meta) {
+    try {
+        repo:get-resource($app, $name)
+        => util:binary-to-string() 
+        => parse-xml()
```

Comment:
The `parse-xml()` function returns a `document-node()`. As a result, the `packages:get-package-meta()` function, whose signature says it returns `element(repo:meta)`, throws an error.

Reply:
I should have made clear that this is still work in progress, sorry.
But on the other hand you already had a look. And the question is not so simple to answer: 
Which route should we take? Remove the return type again or we could also change the body of the try block to
```
 let $metadata-document := repo:get-resource($app, $name)
        => util:binary-to-string() 
        => parse-xml()
 return $metadata-document/repo:meta
```

Reply:
This route works! I appreciate having the more specific return type in the function signature.

Reply:
Sometimes being a little more generic is better :)